packages/server/src/ui/components/lhr-viewer-link.jsx

```diff
@@ -13,7 +13,8 @@ import './lhr-viewer-link.css';
 
 /** @param {LH.Result} lhr */
 export function openLhrInClassicViewer(lhr) {
-  const VIEWER_ORIGIN = 'https://googlechrome.github.io';
+  // Allows for self-hosted viewer uri instead of linking externally
+  const VIEWER_ORIGIN = process.env.VIEWER_ORIGIN != null ? process.env.VIEWER_ORIGIN : 'https://googlechrome.github.io';
```

Comment:
I appreciate the narrowness, but if `VIEWER_ORIGIN` is empty string we'd probably still want to use the default :)

```suggestion
  const VIEWER_ORIGIN = process.env.VIEWER_ORIGIN || 'https://googlechrome.github.io';
```

this will only work though if `VIEWER_ORIGIN` is defined when the project is *built* so the npm package won't work out of the box. Should it accept a URL flag/`localStorage`/`Cookie` as well?

Reply:
> this will only work though if `VIEWER_ORIGIN` is defined when the project is _built_

Couldn't the server tell the frontend about the viewer URL somehow? Maybe a configurable setting per project even with whatever's set for the server via env as default?

Reply:
> Couldn't the server tell the frontend about the viewer URL somehow?

Yes, as suggested, cookies would be one way of doing that, but a more heavy-handed explicit configuration per project could also work (way more work on the UI-side though to fetch the project settings and thread through all the layers that require a report render button). I don't suspect that this is terribly important to configure *per project* either. It's either a closed instance of the server or it's not in most cases.

Reply:
> I don't suspect that this is terribly important to configure *per project* either. It's either a closed instance of the server or it's not in most cases.

True, that might have been an overzealous thought 🙂

Reply:
So, having looked around previously- and not having considered that the environment variable would be not useful as part of the npm package distribution because the UI is built already... maybe it makes more sense to solve this particular piece with #461 instead? If I understand that issue correctly it would pull a component of the lighthouse package into this one, minus the firebase auth.

Reply:
> maybe it makes more sense to solve this particular piece with #461 instead?

It definitely does make more sense, but making this configurable still seems fine. Let's leave it as-is and plan to tackle #461 👍

Reply:
Sounds like a plan, I'm happy to take a crack at it this weekend. Thanks for all the help!

Reply:
Awesome that'd be great! It shouldn't be *too* bad, I'm thinking a single route on the server that accepts a runId and then returns the HTML report as-is from 

https://github.com/GoogleChrome/lighthouse-ci/blob/438f3e1c60ce1b4396024fad01220e7a6e376246/packages/utils/src/saved-reports.js#L53

The harder part will be replacing the existing buttons that link to the viewer to use the new route with `runId` instead.

Reply:
> Awesome that'd be great! It shouldn't be _too_ bad, I'm thinking a single route on the server that accepts a runId and then returns the HTML report as-is from

Wanted to report back that I did end up taking a stab at this over the weekend and got to a point where I _think_ what I have will work, but kept running into issues with the seed data the server generates. It looks like the seed data is based on a pre-release version lighthouse 6 and is no longer compatible with the render code (can be seen with the current external lighthouse viewer from a seed data generated build).

When I find some more time this week I'll go through with adding a test project and generating some lighthouse 8 reports to validate it works and cleaning up the buttons for a PR.